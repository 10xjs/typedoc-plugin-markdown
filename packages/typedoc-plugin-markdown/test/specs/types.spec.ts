import * as Handlebars from 'handlebars';

import { TestApp } from '../test-app';

describe(`Types:`, () => {
  let testApp: TestApp;

  beforeAll(() => {
    testApp = new TestApp(['types.ts']);
  });

  test(`should compile 'array' type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('arrayType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'stringLiteral' type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('stringLiteralType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'union' of string literals types'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('unionType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile intrinsic type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('stringType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'literal' type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('literalType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'tuple' type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('tupleType').type,
      ),
    ).toMatchSnapshot();
  });

  test(`should compile 'intersection' type'`, () => {
    testApp.bootstrap();
    expect(
      TestApp.compileHelper(
        Handlebars.helpers.type,
        testApp.findReflection('intersectionType').type,
      ),
    ).toMatchSnapshot();
  });
});
