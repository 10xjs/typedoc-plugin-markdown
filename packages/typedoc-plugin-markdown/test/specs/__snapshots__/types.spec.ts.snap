// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Types: should compile 'array' type' 1`] = `
"string[]
"
`;

exports[`Types: should compile 'intersection' type' 1`] = `
"[IntersectionClassA](_types_.intersectionclassa.md) & [IntersectionClassB](_types_.intersectionclassb.md)
"
`;

exports[`Types: should compile 'literal' type' 1`] = `
"{ a: string ; b: number  }
"
`;

exports[`Types: should compile 'stringLiteral' type' 1`] = `
"\\"blue\\"
"
`;

exports[`Types: should compile 'tuple' type' 1`] = `
"[string, number]
"
`;

exports[`Types: should compile 'union' of string literals types' 1`] = `
"\\"ease-in\\"  | \\"ease-out\\"  | \\"ease-in-out\\"
"
`;

exports[`Types: should compile intrinsic type' 1`] = `
"string
"
`;
